app/publish/tus.go

```diff
@@ -0,0 +1,204 @@
+package publish
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/lbryio/lbrytv/app/auth"
+	"github.com/lbryio/lbrytv/app/proxy"
+	"github.com/lbryio/lbrytv/app/query/cache"
+	"github.com/lbryio/lbrytv/app/rpcerrors"
+	"github.com/lbryio/lbrytv/app/sdkrouter"
+	"github.com/lbryio/lbrytv/internal/errors"
+	"github.com/lbryio/lbrytv/internal/metrics"
+	"github.com/lbryio/lbrytv/internal/monitor"
+	"github.com/lbryio/lbrytv/internal/responses"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+	tusd "github.com/tus/tusd/pkg/handler"
+	"github.com/ybbus/jsonrpc"
+)
+
+const module = "tus.publish"
+
+// PublishHandler handle media publishing on odysee-api, it implements TUS
+// specifications to support resumable file upload and extends the handler to
+// support fetching media from remote url.
+type PublishHandler struct {
+	*tusd.UnroutedHandler
+
+	logger       monitor.ModuleLogger
+	composer     *tusd.StoreComposer
+	authProvider auth.Provider
+}
+
+// NewHandler creates a new publish handler.
+func NewHandler(authProvider auth.Provider, cfg tusd.Config, uploadPath string) (*PublishHandler, error) {
+	h := &PublishHandler{}
+
+	if authProvider == nil {
+		return nil, fmt.Errorf("auth provider cannot be nil")
+	}
+
+	// make sure target directory is exist
```

Comment:
While I'm strongly in favor of documenting and commenting the code, I would argue this comment is unnecessary as it is fairly obvious what `os.MkdirAll` does. Same goes for `tusd.NewUnroutedHandler` call below.